samples/Event_Jsvm/app/application.cpp

```diff
@@ -67,21 +69,38 @@ bool triggerEvent(const String& name, const JS::Object& params)
 		return false;
 	}
 
-	// Build the event object...
-	JS::Object event;
-	event["name"] = name;
-	event["params"] = params;
+	JS_TRY()
```

Comment:
@mikee47 I am wondering isn't it possible to use try-catch directly in the JavaScript code and catch out-of-memory exception? We could expose a `reset()` JavaScript function that resets the VM and give the JS developers (end user) better control. On the C/C++ side we could check if the VM is in error state and restart the VM automatically if it stays in this error state for 8 seconds.

Reply:
Unfortunately fatal errors are not the same as javascript exceptions and cannot be caught in script. The engine is essentially halted. No attempt to even tidy up intermediate allocations so cleanup won't work. Try running the tests with `JERRY_ENABLE_DEBUG=` and see what happens!

A `reset()` function would be very easy to implement. However, depending on the use-case if we wanted multiple clients runninng together would we really want one of them to be able to pull the rug out from the others? Maybe, but that'll depend on usage.